packages/at_client/test/self_key_encryption_test.dart

```diff
@@ -0,0 +1,72 @@
+import 'package:at_chops/at_chops.dart';
+import 'package:at_client/src/decryption_service/self_key_decryption.dart';
+import 'package:at_client/src/encryption_service/self_key_encryption.dart';
+import 'package:at_commons/at_builders.dart';
+import 'package:test/test.dart';
+import 'package:mocktail/mocktail.dart';
+import 'package:at_client/at_client.dart';
+import 'package:at_lookup/at_lookup.dart';
+
+class MockAtClientImpl extends Mock implements AtClient {}
+
+class MockLocalSecondary extends Mock implements LocalSecondary {}
+
+class MockRemoteSecondary extends Mock implements RemoteSecondary {}
+
+class MockAtLookupImpl extends Mock implements AtLookUp {}
+
+class FakeLocalLookUpVerbBuilder extends Fake implements LLookupVerbBuilder {}
+
+void main() {
+  AtClient mockAtClient = MockAtClientImpl();
+  AtLookUp mockAtLookUp = MockAtLookupImpl();
+  LocalSecondary mockLocalSecondary = MockLocalSecondary();
+  RemoteSecondary mockRemoteSecondary = MockRemoteSecondary();
+  setUp(() {
+    reset(mockAtLookUp);
+    when(() => mockAtClient.getLocalSecondary())
+        .thenAnswer((_) => mockLocalSecondary);
+    when(() => mockAtClient.getRemoteSecondary())
+        .thenAnswer((_) => mockRemoteSecondary);
+
+    registerFallbackValue(FakeLocalLookUpVerbBuilder());
+  });
+
+  test('test to check self key encryption', () async {
```

Comment:
Please add a brief description of what this test is testing

Reply:
done